File: C++/move-zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int i = 0, j = 1, sz = nums.size();
        
        while (j < sz) {
            if (nums[i] == 0) {
                if (nums[j] != 0) {
                    nums[i] = nums[j];
                    nums[j] = 0;
                    i++;
                    j++;
                }
                else {
                    j++;
                }
            }
            else {
                i++;
                j++;
            }
        }
    }
};

File: C#/move-zeroes.cs
public class Solution {
    public void MoveZeroes(int[] nums) {
        int dest = 0;
        for (int i = 0; i < nums.Length; i++) {
            if (nums[i] != 0) {
                nums[dest++] = nums[i];
            }
        }
        while (dest < nums.Length) {
            nums[dest++] = 0;
        }
    }
}